document/src/test/java/com/yahoo/document/update/SerializationTestCase.java

```diff
@@ -57,4 +65,29 @@ public void testClearSerialization() {
         assertEquals("'field1' [clear]", deserializedUpdate.toString());
     }
 
+    private static TensorFieldValue createTensor(TensorType type, String tensorCellString) {
+        return new TensorFieldValue(Tensor.from(type, tensorCellString));
+    }
+
+    private GrowableByteBuffer serializeUpdate(FieldUpdate update) {
+        DocumentSerializer buffer = DocumentSerializerFactory.createHead(new GrowableByteBuffer());
+        update.serialize(buffer);
+        buffer.getBuf().rewind();
+        return buffer.getBuf();
+    }
+
+    private FieldUpdate deserializeUpdate(GrowableByteBuffer buffer) {
+        return new FieldUpdate(DocumentDeserializerFactory.createHead(new DocumentTypeManager(), buffer), documentType, Document.SERIALIZED_VERSION);
+    }
+
+    @Test
+    public void testTensorModifySerialization() {
+        FieldUpdate update = new FieldUpdate(tensorField);
+        TensorFieldValue tensor = createTensor(tensorType, "{{x:8,y:9}:2}");
+        update.addValueUpdate(new TensorModifyUpdate(TensorModifyUpdate.Operation.REPLACE, tensor));
+        GrowableByteBuffer buffer = serializeUpdate(update);
+        FieldUpdate deserializedUpdate = deserializeUpdate(buffer);
+        assertEquals("'tensorfield' [tensormodify replace TensorFieldValue {\n    classId: 1\n}\n]", deserializedUpdate.toString());
```

Comment:
I would check the operation and tensor separately instead of depending on toString() format.